sentry_sdk/integrations/huey.py

```diff
@@ -0,0 +1,148 @@
+from __future__ import absolute_import
+
+import sys
+from datetime import datetime
+
+from sentry_sdk._compat import reraise
+from sentry_sdk._types import MYPY
+from sentry_sdk import Hub
+from sentry_sdk.integrations import DidNotEnable, Integration
+from sentry_sdk.tracing import Transaction, TRANSACTION_SOURCE_TASK
+from sentry_sdk.utils import capture_internal_exceptions, event_from_exception
+
+if MYPY:
+    from typing import Any, Callable, Optional, Union, TypeVar
+
+    from sentry_sdk._types import EventProcessor, Event, Hint
+    from sentry_sdk.utils import ExcInfo
+
+    F = TypeVar("F", bound=Callable[..., Any])
+
+try:
+    from huey.api import Huey, Result, ResultGroup, Task
+    from huey.exceptions import CancelExecution, RetryTask
+except ImportError:
+    raise DidNotEnable("Huey is not installed")
+
+
+HUEY_CONTROL_FLOW_EXCEPTIONS = (CancelExecution, RetryTask)
+
+
+class HueyIntegration(Integration):
+    identifier = "huey"
+
+    @staticmethod
+    def setup_once():
+        # type: () -> None
+        patch_enqueue()
+        patch_execute()
+
+
+def patch_enqueue():
+    # type: () -> None
+    old_enqueue = Huey.enqueue
+
+    def _sentry_enqueue(self, task):
+        # type: (Huey, Task) -> Optional[Union[Result, ResultGroup]]
+        hub = Hub.current
+
+        if hub.get_integration(HueyIntegration) is None:
+            return old_enqueue(self, task)
+
+        with hub.start_span(op="huey.enqueue", description=task.name):
```

Comment:
Oh and please add `OP.QUEUE_SUBMIT_HUEY` constant (set to "queue.submit.huey") here: https://github.com/getsentry/sentry-python/blob/master/sentry_sdk/consts.py#L110 and then use the constant.